src/all/mitaku/src/eu/kanade/tachiyomi/extension/all/mitaku/Mitaku.kt

```diff
@@ -0,0 +1,187 @@
+import org.jsoup.nodes.Element
+import rx.Observable
+
+class Mitaku : ParsedHttpSource() {
+    override val name = "Mitaku"
+
+    override val baseUrl = "https://mitaku.net"
+
+    override val lang = "all"
+
+    override val supportsLatest = false
+
+    // ============================== Popular ===============================
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/category/ero-cosplay/page/$page", headers)
+
+    override fun popularMangaSelector() = "div.article-container article"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+
+        title = element.selectFirst("a")!!.attr("title")
+
+        thumbnail_url = element.selectFirst("img")?.absUrl("src")
+    }
+
+    override fun popularMangaNextPageSelector() = "div.wp-pagenavi a.page.larger"
+
+    // =============================== Latest ===============================
+    override fun latestUpdatesRequest(page: Int): Request {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesSelector(): String {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesFromElement(element: Element): SManga {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesNextPageSelector(): String? {
+        throw UnsupportedOperationException()
+    }
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val filterList = if (filters.isEmpty()) getFilterList() else filters
+        val tagFilter = filterList.findInstance<TagFilter>()!!
+        val categoryFilter = filterList.findInstance<CategoryFilter>()!!
+
+        return when {
+            query.isEmpty() && categoryFilter.state != 0 -> {
+                val url = baseUrl.toHttpUrl().newBuilder()
+                    .addPathSegment("category")
+                    .addPathSegment(categoryFilter.toUriPart())
+                    .addPathSegment("page")
+                    .addPathSegment(page.toString())
+                    .build()
+                GET(url)
+            }
+            query.isEmpty() && tagFilter.state.isNotEmpty() -> {
+                val url = baseUrl.toHttpUrl().newBuilder()
+                    .addPathSegment("tag")
+                    .addPathSegment(tagFilter.toUriPart())
+                    .addPathSegment("page")
+                    .addPathSegment(page.toString())
+                    .build()
+                GET(url)
+            }
+            query.isNotEmpty() -> {
+                val url = baseUrl.toHttpUrl().newBuilder()
+                    .addPathSegment("page")
+                    .addPathSegment(page.toString())
+                    .addQueryParameter("s", query)
+                    .build()
+                GET(url)
```

Comment:
You need to add headers here, and elsewhere:
```suggestion
                GET(url, headers)
```